browser/src/components/LogView/BranchGraph/index.tsx

```diff
@@ -259,25 +267,87 @@ function drawGitGraph(svg: SVGSVGElement, content: HTMLElement, startAt: number,
             }
         }
     }
+
+    branches.forEach(branch => {
+        const svgPath = branch.path.cmds + currentY;
+        branch.path.setAttribute('d', svgPath);
+        svgPaths.set(branch.path, svgPath);
+    });
+
+    debugger;
+    const lines: Point[][] = [];
+    svgPaths.forEach((pathOrSvg, svg) => {
+        try {
+            lines.push(getPointsFromPath(pathOrSvg));
+        } catch (ex) {
+            console.error('Failed to generate SVG line path', ex);
+        }
+    });
+
+    // Sort all points in lines.
+    lines.forEach((points, i) => {
+        lines[i] = points.sort((a, b) => a.y - b.y);
+    });
     for (let i = startAt; i < content.children.length; ++i) {
         let element = content.children[i];
         if (i >= entries.length) {
             break;
         }
         let minLeft = Math.min(maxLeft, 3);
         let left = element ? Math.max(minLeft, (element as any).branchesOnLeft) : minLeft;
-        element.setAttribute('style', element.getAttribute('style') + ';padding-left:' + (left + 1) * lastXOffset + 'px');
-    }
-    branches.forEach(branch => {
-        branch.path.setAttribute('d', branch.path.cmds + currentY);
-    });
+        const originalStyle = element.getAttribute('style');
+        element.setAttribute('style', originalStyle + ';padding-left:' + (left + 1) * lastXOffset + 'px');
+        try {
+            const pointsAtY = lines.map(points => getPointAtY((i + 1) * logEntryHeight, points));
```

Comment:
I want the old code to work, this is a hacky improvement. If this bombs out, at least the old approach works.